### PR TITLE
Add water particle free energy datasets

### PR DESCRIPTION
**Pull request by harlor**
As suggested in https://github.com/alchemistry/alchemlyb/pull/62 I want to add 3 new free energy data sets:

- One without an energy column
- One with a potential energy column
- One with a total energy column

These data sets are useful to test the calculation of the reduced potential for gromacs data sets in alchemlyb. See https://github.com/alchemistry/alchemlyb/issues/59.

### CODE REVIEW

File: src/alchemtest/gmx/access.py [hunk 1]
```diff
@@ -102,3 +102,72 @@ def load_expanded_ensemble_case_3():
...
+        Dictionary-like object, the interesting attributes are:
+
+        - 'data' : the data files by alchemical leg
+        - 'DESCR': the full description of the dataset
+
+    """
+
+    module_path = dirname(__file__)
+
+    data = {'AllStates': glob(join(module_path, 'water_particle/without_energy/*.xvg.bz2'))}
```
Comment (orbeckst) [inline src/alchemtest/gmx/access.py]: Don't use Unix path separators but let `join` do everything:
```python
glob(join(module_path, 'water_particle', 'without_energy', '*.xvg.bz2'))
```
to make sure it also works on Windows.

File: src/alchemtest/gmx/access.py [hunk 2]
```diff
@@ -102,3 +102,72 @@ def load_expanded_ensemble_case_3():
...
+        Dictionary-like object, the interesting attributes are:
+
+        - 'data' : the data files by alchemical leg
+        - 'DESCR': the full description of the dataset
+
+    """
+
+    module_path = dirname(__file__)
+
+    data = {'AllStates': glob(join(module_path, 'water_particle/with_potential_energy/*.xvg.bz2'))}
```
Comment (orbeckst) [inline src/alchemtest/gmx/access.py]: Don't use Unix path separators but let `join` do everything:

File: src/alchemtest/gmx/access.py [hunk 3]
```diff
@@ -102,3 +102,72 @@ def load_expanded_ensemble_case_3():
...
+        Dictionary-like object, the interesting attributes are:
+
+        - 'data' : the data files by alchemical leg
+        - 'DESCR': the full description of the dataset
+
+    """
+
+    module_path = dirname(__file__)
+
+    data = {'AllStates': glob(join(module_path, 'water_particle/with_total_energy/*.xvg.bz2'))}
```
Comment (orbeckst) [inline src/alchemtest/gmx/access.py]: Don't use Unix path separators but let `join` do everything:

File: src/alchemtest/gmx/water_particle/descr.rst [hunk 4]
```diff
@@ -0,0 +1,27 @@
+Gromacs: water particle
+=========================
+
+Free energy Estimation of a water particle between to hydrophilic surfaces
```
Comment (orbeckst) [inline src/alchemtest/gmx/water_particle/descr.rst]: estimation

File: src/alchemtest/gmx/access.py (hunk 1 above)
Comment (harlor) [reply src/alchemtest/gmx/access.py]: Oh I wasn't aware of that. I guess this should also be fixed in the other accessors functions.

I'll create a separate PR for that.

File: src/alchemtest/gmx/water_particle/descr.rst (hunk 4 above)
Comment (harlor) [reply src/alchemtest/gmx/water_particle/descr.rst]: fixed.

File: src/alchemtest/gmx/access.py (hunk 1 above)
Comment (orbeckst) [reply src/alchemtest/gmx/access.py]: Yes, I am sure. Sorry, sometimes the new PR is the one that reveals a general shortcoming in old code...

**Comment by orbeckst** [comment]
I approved a bit prematurely because the entries in `gmx.rst` are still missing. But just ping me when you have added it and I'll merge.

**Comment by harlor** [comment]
Ok - I added the information to the gmx.rst

**Comment by orbeckst** [comment]
Thank you!

**Pull request merged**